core/execution/future/liquidity_provision.go

```diff
@@ -35,998 +26,33 @@ func (m *Market) SubmitLiquidityProvision(
 	ctx context.Context,
 	sub *types.LiquidityProvisionSubmission,
 	party, deterministicID string,
-) (err error,
-) {
+) error {
 	defer m.onTxProcessed()
 
-	m.idgen = idgeneration.New(deterministicID)
-	defer func() { m.idgen = nil }()
-
-	if !m.canSubmitCommitment() {
-		return common.ErrCommitmentSubmissionNotAllowed
-	}
```

Comment:
we probably still want this.

Reply:
unless it's done in the liquidity engine?

Reply:
Correct. It is in the market liquidity